Tests/E2E/ClassCTest.cs

```diff
@@ -54,25 +55,11 @@ public async Task Test_ClassC_Send_Message_From_Direct_Method_Should_Be_Received
                 RawPayload = Convert.ToBase64String(new byte[] { 0xFF, 0x00 }),
             };
 
-            TestLogger.Log($"[INFO] Using service client to call direct method to {TestFixture.Configuration.LeafDeviceGatewayID}/{TestFixture.Configuration.NetworkServerModuleID}");
+            TestLogger.Log($"[INFO] Using service API to send C2D message to device {device.DeviceID}");
             TestLogger.Log($"[INFO] {JsonConvert.SerializeObject(c2d, Formatting.None)}");
 
-            for (var i = 0; i < MAX_MODULE_DIRECT_METHOD_CALL_TRIES; ++i)
```

Comment:
As per our offline discussion: I would argue that if we do not want to deprecate the direct method path for the user (and always force the user to go through the function endpoint), we should keep the old test to validate the direct method path directly.

The benefits I see from this are:
- The E2E test explicitly asserts a guarantee a functionality that we provide for the user (the direct method may still be used without going through the function)
- The direct method path will not only be implicitly covered, but also explicitly covered. If we ever decide to change the function implementation, we will still cover the direct method path.

Reply:
I see the point of verifying that the direct method path works but at the same time I don't really like the idea of having the test execute essentially the same logic twice (the endpoint path is also using the direct method under the hood). How about instead having another assertion in the test that LNS received the direct method call (e.g. asserting the log line: `[TCPLOG] [itestarm1] 01F5A90000000024: received cloud to device message from direct method: {"DevEUI":"01F5A90000000024","Fport":23,"RawPayload":"/wA=","Payload":null,"Confirmed":false,"MessageId":"7c210b2e-4bb9-43c3-8fea-1a1eec03c8f4","MacCommands":[]}` which is logged by the LNS?
Does anyone else have an opinion here - @p-schuler @danigian

Reply:
From my point of view the E2E tests should test the user behavior. If, in the documentation, we specify that the way for sending the C2D messages is through the endpoint, that's what we should test. Adding an additional assertion as Maggie is suggesting would be nice.

If we want to keep the explicit testing of the Direct Method, I would rather write an integration test to actually verify that it's properly invoked and handled (if that does not already exists).

No strong opinion here 👍

Reply:
In that case let's leave the E2E test simply using the function endpoint. I added the assertion checking that the direct method usage is logged in the LNS.